assets/test/e2e/test.js

```diff
@@ -33,9 +33,100 @@ describe('', function() {
 
     driver.findElement(by.name("username")).sendKeys("e2e-user");
     driver.findElement(by.name("password")).sendKeys("e2e-user");
-    driver.findElement(by.css("button[type='submit']")).click();
+    return driver.findElement(by.css("button[type='submit']")).click();
   };
 
+  var goToAddToProjectPage = function(projectName) {
+    var uri = '/project/' + projectName + '/create';
+    browser.get(uri);
+    expect(element(by.cssContainingText('h1', "Create Using Your Code")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('h1', "Create Using a Template")).isPresent()).toBe(true);
+    expect(element(by.model('from_source_url')).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('.catalog h3 > a', "ruby-helloworld-sample")).isPresent()).toBe(true);
+  }
+
+  var goToCreateProjectPage = function() {
+    browser.get('/createProject');
+    expect(element(by.cssContainingText('h1', "New Project")).isPresent()).toBe(true);
+    expect(element(by.model('name')).isPresent()).toBe(true);
+    expect(element(by.model('displayName')).isPresent()).toBe(true);
+    expect(element(by.model('description')).isPresent()).toBe(true);
+  }
+
+  var requestCreateFromSource = function(projectName, sourceUrl) {
+    var uri = '/project/' + projectName + '/create';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    element(by.model('from_source_url')).clear().sendKeys(sourceUrl);
+    var nextButton = element(by.buttonText('Next'));
+    browser.wait(protractor.ExpectedConditions.elementToBeClickable(nextButton), 2000);
+    return nextButton.click();
+  }
+
+  var requestCreateFromTemplate = function(projectName, templateName) {
+    var uri = '/project/' + projectName + '/create';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    var template = element(by.cssContainingText('.catalog h3 > a', templateName));
+    expect(template.isPresent()).toBe(true);
+    return template.click();
+  }
+
+  var attachBuilderImageToSource = function(projectName, builderImageName) {
+    var uri = '/project/' + projectName + '/catalog/images';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    expect(element(by.cssContainingText('h1', "Select a builder image")).isPresent()).toBe(true);
+    var builderImageLink = element(by.cssContainingText('div > h3 > a', builderImageName));
```

Comment:
do you need the 'div >' portion of the selector, will it find it with just h3 > a ?  Want to make these tests as robust to change as possible